sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -799,7 +799,7 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
           }
 
           intercept[NoSuchDatabaseException] {
-            sql(s"ALTER DATABASE `db-not-exist` SET LOCATION '${tmpDir.toURI}'")
+            sql(s"ALTER DATABASE `db_not_exist` SET LOCATION '${tmpDir.toURI}'")
```

Comment:
I think you only add a check for catalog, why it affects how the database resolves?

Reply:
@viirya The testcase is `ALTER DATABASE `db-not-exist` SET LOCATION '${tmpDir.toURI}` , so it will call `visitSetNamespaceLocation` first. Then `UnresolvedNamespace` to `ResolvedNamespace`, it will call the the rule `ResolveNamespace`. It goes to the last case branch ```scala
case UnresolvedNamespace(CatalogAndNamespace(catalog, ns)) =>
      ResolvedNamespace(catalog, ns)
```
In `CatalogAndNamespace` it will call `Some((catalogManager.catalog(nameParts.head), nameParts.tail))`, I add the check logic in method `catalogManager.catalog`.

In the test case , The nameParts.length == 1 and nameParts.head == `db-not-exist`, The name is not valid, so it will throw a AnalysisException. The original logic will throw a `CatalogNotFoundException`, then use `currentCatalog`, `db-not-exist` will as a name of DataBase.

Maybe I'm wrong ?

Reply:
@viirya 
The testcase is `ALTER DATABASE `db-not-exist` SET LOCATION '${tmpDir.toURI}` , so it will call `visitSetNamespaceLocation` first. 
Then `UnresolvedNamespace` to `ResolvedNamespace`, it will call the the rule `ResolveNamespace`. It goes to the last case branch.

 ```scala
case UnresolvedNamespace(CatalogAndNamespace(catalog, ns)) =>
      ResolvedNamespace(catalog, ns)
```

In `CatalogAndNamespace` it will call `Some((catalogManager.catalog(nameParts.head), nameParts.tail))`, I add the check logic in method `catalogManager.catalog`.

In the test case , The nameParts.length == 1 and nameParts.head == `db-not-exist`, The name is not valid, so it will throw a AnalysisException. The original logic will throw a `CatalogNotFoundException`, then use `currentCatalog`, `db-not-exist` will as a name of DataBase.

Maybe I'm wrong ?

Reply:
Seems `validateName` was only used for validation when creating database, table before. So previously `ALTER DATABASE db-not-exist` didn't trigger the validation. It just simply checks the existing of the database `db-not-exist`.

So after this change, `ALTER DATABASE db-not-exist` will trigger catalog name check even `db-not-exist` here doesn't indicate catalog name but database name.

That being said, a command like ``ALTER DATABASE db-not-exist` will throw the `AnalysisException`, not `NoSuchDatabaseException` like before.

What happens if the database name is with catalog? As we don't validate database name for this, if it throws `NoSuchDatabaseException`, seems not consistent.

Reply:
In this testcase, if there is a catalog named `db-not-exist` that same with table name `db-not-exist`, we can not find the database `db-not-exist` that it exist actually.

Reply:
Maybe we can't allow catalog's name to appear in the sql statement. end-uses can switch catalogs only by using the `use` statement ?

Reply:
To be clear, `ALTER DATABASE db-not-exist` previously throws `NoSuchDatabaseException`, right?

After this change, as `db-not-exist` will be taken in catalog resolving `CatalogAndNamespace`, it will throw `AnalysisException`, right? So you changed the test to `db_not_exist` so it won't fail catalog name validation.

So what I mean is, now `ALTER DATABASE db-not-exist` throws `AnalysisException`, due to `db-not-exist` fails catalog name validation. But what if `ALTER DATABASE catalog.db-not-exist`? Does it still throw `NoSuchDatabaseException` like before?

If so, the two commands are pretty close or the same, as they both ask to alter a non-existing database. But one throws `AnalysisException`, and another one throws `NoSuchDatabaseException`?

I don't run it to validate. But from above description, seems it is what will happen?

Reply:
> To be clear, `ALTER DATABASE db-not-exist` previously throws `NoSuchDatabaseException`, right?
> 
right.

> After this change, as `db-not-exist` will be taken in catalog resolving `CatalogAndNamespace`, it will throw `AnalysisException`, right? So you changed the test to `db_not_exist` so it won't fail catalog name validation.
> 
right.

> So what I mean is, now `ALTER DATABASE db-not-exist` throws `AnalysisException`, due to `db-not-exist` fails catalog name validation. But what if `ALTER DATABASE catalog.db-not-exist`? Does it still throw `NoSuchDatabaseException` like before?
> 
if `ALTER DATABASE catalog.db-not-exist`. it still throw `NoSuchDatabaseException` like before.

> If so, the two commands are pretty close or the same, as they both ask to alter a non-existing database. But one throws `AnalysisException`, and another one throws `NoSuchDatabaseException`?
> 
right.

> I don't run it to validate. But from above description, seems it is what will happen?
You are right.